src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -3672,6 +3672,7 @@ Copyright (C) Microsoft Corporation. All rights reserved.
     <ItemGroup>
       <CustomAdditionalCompileInputs Include="$(IntermediateOutputPath)$(MSBuildProjectFile).CoreCompileInputs.cache" />
       <CoreCompileCache Include="@(Compile)" />
+      <CoreCompileCache Include="@(EmbeddedResource)" />
```

Comment:
`@(EmbeddedResource)` is not passed to the compiler so it's not the right item here. It looks like `@(_CoreCompileResourceInputs)` would be the right item:

https://github.com/dotnet/roslyn/blob/990f487dd477db0fecb14ab5aa4f0e66f416a437/src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets#L131

That's defined here:

https://github.com/dotnet/msbuild/blob/46d8f9b0bbda55fd6e72a5bdfcbd7dbf0b420839/src/Tasks/Microsoft.Common.CurrentVersion.targets#L3503-L3505

So this PR should probably take a dependency on that target too.

Reply:
I see, so we extract some outputs of `EmbeddedResource` items based on whether they are `Resx` or not! Only those are passed into `CoreCompile`.

If that's the case, then, shouldn't we do the same for `ReferencePath`? Here, the `ReferencePathWithRefAssemblies` items are the final input to the `CoreCompile`, not `ReferencePath` items.

You also have other items as inputs too. Should we include any of them as well?

Reply:
I see, so we extract some outputs of `EmbeddedResource` items based on whether they are `Resx` or not without Culture! Only those are passed into `CoreCompile`, the rest is ignored.

If that's the case, then, shouldn't we do the same for `ReferencePath`? Here, the `ReferencePathWithRefAssemblies` items are the final input to the `CoreCompile`, not `ReferencePath` items.

You also have other items as inputs too. Should we include any of them as well?

Reply:
Probably a good idea. We should also get a review from the compiler team to make sure what we end up going with makes sense.

Reply:
Hard to comment here cause I'm unsure exactly what this part of MSBuild is achieving. Hence can't say what is and is not good items to be using here.

Reply:
As I understand it, the FUTD check looks at the CoreCompileCache to see what items need to be up-to-date for it to be able to skip executing CoreCompile. Currently CoreCompileCache is missing things that are actually used by CoreCompile, so if you update EmbeddedResources, the FUTD check says it's up-to-date and skips CoreCompile anyway. This aims to fix that, but we'd also like to hit other things that, when changed, should make CoreCompile run again that are currently missed.

Reply:
This does not have anything to do with FUTD, which is a project system concept.

This target tries to determine whether the _set_ of inputs to `CoreCompile` has changed since the last time the compiler was run. It was added because of a scenario new to globbing support:

1. Build project.
2. Delete source file `foo.cs`.
3. Build again
4. `CoreCompile` thinks it's up to date because the output `.dll` is newer than all inputs, but that's wrong because an input was removed.

Prior to extensive use of globbing, the `$(MSBuildAllProjects)` input to `CoreCompile` caught this case because the project file was modified to remove the source file. But with globbing, we needed a build-time step.

This PR is extending this from `@(Compile)` items to include `@(EmbeddedResource)` which has a similar problem--we could skip running the compiler and leave an output with an extra embedded resource.

The question is: are there other inputs likely derived from globs in the SDK case that we should consider here?

Reply:
That's my question as well… <https://github.com/dotnet/msbuild/issues/5334#issuecomment-879350699>

From what I see, we have modules, embedded files, strong name key and Win32 manifest, Icon and resources!

Not to mention WPF items and its 2-pass  compilation! _That's on a whole 'nother level!!_

Reply:
> The question is: are there other inputs likely derived from globs in the SDK case that we should consider here?

What all can globs be used for now? Should my mental model just be anything on disk? If so then probably need `$(AdditionalFiles)`, references and `$(EmbeddedFiles)` at least.

I omitted `$(EditorConfigFiles)` because I think those are handled specially in MSBuild already. 

@RikkiGibson, @chsienki, @cston